File: 2D-Matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
       if(matrix == null || matrix.length == 0 || matrix[0].length == 0) {
           return false;
       }
        int m = matrix.length; //row
        int n = matrix[0].length; //column
        
        int i = 0;
        int j = n - 1;
        
        
        
        
        
         while(i < m && j >= 0) {
            if (matrix[i][j] == target) {
                return true;
            }
            else if (matrix[i][j] < target) {
                i += 1;
            }
            
            else {
                j -=1;
            }        }
        return false;
    }
}

File: 2D Matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        ###################
        # TC - o(m+n)
        # SC -  o(1)
        ###################
        m = len(matrix)
        n = len(matrix[0])
        # base case
        if m == 0:
            return False
        # the aproach is to travel towards the direction of element 
        # start with right corner
        r = 0
        c = n - 1
        while(r<m and  c>=0):
            # print(r)
            if matrix[r][c] == target:
                return True
            elif target > matrix[r][c]:
                r= r + 1
            else:
                c = c - 1
        return False
